### PR TITLE
ci: switch back to executing functional tests with powershell

### PR DESCRIPTION
**Pull request by m3dwards**
In this PR: https://github.com/bitcoin/bitcoin/pull/29535 it was noticed that the functional test runner was skipping running the tests with the message of `Test '' not found in full test list`

I believe that this was caused by Powershell passing a blank `TEST_RUNNER_EXTRA` environment variable as an empty flag to the `test_runner.py` script which then tries to find a test to run with a blank name.

This PR switches the functional tests back to using Powershell but instead of passing an empty argument, constructs a command string and then executes that command. It also prints the powershell version to help diagnose any issues in the future (not that this issue was related to powershell version).

I believe it's best to switch back to powershell rather than stick with cmd, even though it currently works, as this is what I would expect the majority of users and windows developers would use.

---

Some extra notes:

- Run with `--extended takes` ~ 28 minutes
- Run without `--extended takes` ~ 15 minutes
- The mac job appears to never run with `--extended`

### CODE REVIEW

**Comment by fanquake** [comment]
Seems fine, however I still don't understand what/how this broke in the first place. Givne the timings you posted above, I'm somewhat inclined for us to just always run with the extended tests, given the only reason this distinction existed was because the runtimes used to much longer. However, ~30 minutes for a run with extended tests seems well within the bounds of all the other CIs, and Native Windows is not a priority-to-finish for developers.

**Comment by m3dwards** [comment]
I think it must have been a python or more likely powershell update that treats empty arguments differently?

**Comment by hebasto** [comment]
> I believe it's best to switch back to powershell rather than stick with cmd, even though it currently works, as this is what I would expect the majority of users and windows developers would use.

@sipsorcery What do you think?

**Comment by hebasto** [comment]
> ... I still don't understand what/how this broke in the first place.

The broken change was introduced in [PowerShell 7.3](https://devblogs.microsoft.com/powershell/powershell-7-3-general-availability/#improvement-to-native-command-argument-passing):
> Windows and Linux/macOS have fundamental differences in how native command arguments are handled specifically when quotes are involved. We added a new feature [`$PSNativeCommandArgumentPassing`](https://learn.microsoft.com/powershell/scripting/learn/experimental-features#psnativecommandargumentpassing) to control how PowerShell passes arguments to native commands. The default behavior for Windows and Linux/macOS should work as most users expect in their respective environments.

**tl;dr:** Empty strings are preserved as arguments now.

The CI was broken due to the PowerShell [upgrade](https://github.com/actions/runner-images/issues/9115) from 7.2.18 to 7.4.1.

**Comment by sipsorcery** [comment]
> > I believe it's best to switch back to powershell rather than stick with cmd, even though it currently works, as this is what I would expect the majority of users and windows developers would use.
> 
> @sipsorcery What do you think?

No strong preference. I use DOS or PowerShell depending on the task at hand. 

I do lean slightly towards DOS as it's simpler. Perhaps the original issue is a proof in point. DOS is unlikely to ever change and break the tests whereas a future PowerShell update might.

**Comment by m3dwards** [comment]
@hebasto great find!

Happy to stick with cmd.exe which I agree is less likely to have breaking changes in the future.

As per @fanquake's suggestion, shall we keep it simple and just run extended always?

**Comment by hebasto** [comment]
> Happy to stick with cmd.exe which I agree is less likely to have breaking changes in the future.

I agree. Besides, we also use the `cmd` shell in the "Build" step.

**Comment by hebasto** [comment]
> As per @fanquake's suggestion, shall we keep it simple and just run extended always?

See the previous discussions in:
- https://github.com/bitcoin/bitcoin/pull/28173#discussion_r1302929493
- https://github.com/bitcoin/bitcoin/pull/28360

**Comment by fanquake** [comment]
> See the previous discussions in:

Can you point to something specific? Not sure that's relevant given the change in runtimes.

**Comment by hebasto** [comment]
> > See the previous discussions in:
> 
> Can you point to something specific? Not sure that's relevant given the change in runtimes, and now they no-longer seem to be randomly failing so often.

> However, ~30 minutes for a run with extended tests seems well within the bounds of all the other CIs, and Native Windows is not a priority-to-finish for developers.

The number of CI Windows runners is limited. Extending runtime will decrease the availability of runners for all developers. Besides, the developer, who is interested in extended functional tests for their PRs, might run GHA CI in their own repo.

**Comment by m3dwards** [comment]
Not much appetite to switch back to powershell so closing this PR.
